frontend/src/components/Dropdown/components/SelectMenu.tsx

```diff
@@ -9,19 +9,63 @@ import { itemToValue } from '../utils/itemUtils'
 import { DropdownItem } from './DropdownItem'
 import { useSelectPopover } from './SelectPopover'
 
+const ListItems = () => {
+  const { items } = useSelectContext()
+  return (
+    <>
+      {items.map((item, idx) => {
+        return (
+          <DropdownItem
+            key={`${itemToValue(item)}${idx}`}
+            item={item}
+            index={idx}
+          />
+        )
+      })}
+    </>
+  )
+}
+
+const NothingFoundItem = () => {
+  const { styles, nothingFoundLabel } = useSelectContext()
+  return (
+    <ListItem role="option" sx={styles.emptyItem}>
+      {nothingFoundLabel}
+    </ListItem>
+  )
+}
+
+const RenderIfOpen = ({
```

Comment:
is this needed 😅 cause we can just {isOpen ? ... : null} below right? or is it for readability sake?

Reply:
Purely for readability sake. Having too many ternary operations in JSX is difficult to read. The current code is already having 2 nested ternary operations.